src/Illuminate/Validation/Validator.php

```diff
@@ -255,6 +255,8 @@ public function parseData(array $data)
             // array based validation rules plus Arr::dot later in the validations.
             if (Str::contains($key, '.')) {
                 $newData[str_replace('.', '->', $key)] = $value;
+            } elseif (Str::contains($key, '*')) {
```

Comment:
Why do we need this? Faster to always do the replace?

Reply:
Yeah I just don't like to refactor in a PR that proposes a change. It's easier to review such PRs since it only contains the changes needed. Might send another PR to update the replacements and remove the checks.

Reply:
I don't understand what you mean? I'm suggesting you change your PR to contain even fewer changes...

Reply:
```php
if (Str::contains($key, '.')) {
    $newData[str_replace('.', '->', $key)] = $value;
} else {
    $newData[str_replace('*', '__asterisk__', $key)] = $value;
}
```

Reply:
the check for the dot has to go as well

Reply:
But removing that actually changes the behaviour?

Reply:
My version and your version behave exactly the same. If you remove the if check on the dot, then the behaviour changes.

Reply:
Why don't we want to replace asterisks if there's also a period in the same key?

Reply:
I'm not sure. If we do want that, then this PR has a bug that the tests don't show.